metals/src/main/scala/scala/meta/internal/rename/RenameProvider.scala

```diff
@@ -128,6 +128,15 @@ final class RenameProvider(
           occ.symbol.isLocal || foundName.contains(realName)
         }
 
+        def shouldCheckImplementation(
+            symbol: String,
+            path: AbsolutePath,
+            textDocument: TextDocument
+        ) =
+          !symbol.desc.isType && !(symbol.isLocal && implementationProvider
+            .defaultSymbolSearch(path, textDocument)(symbol)
+            .exists(info => info.isTrait || info.isClass))
```

Comment:
We should also include: `info.isObject`, `info.isPackageObject`, `info.isEnum`

Reply:
PR is merged and closed, I'll do t in another one